nixos/default.nix

```diff
@@ -23,6 +23,15 @@ let
     };
   });
 
+  serviceEnvironment =
+    optionalAttrs (cfg.backupExt != null) {
+      HOME_MANAGER_BACKUP_EXT = cfg.backupExt;
+    } //
+    optionalAttrs cfg.verbose {
+      VERBOSE_ARG = "--verbose";
```

Comment:
I don't think this is needed, the `VERBOSE_ARG` option is automatically set if `VERBOSE` is set:
https://github.com/rycee/home-manager/blob/31ae1bc2fff9660a1870161e064e86d3b4d6b396/modules/lib-bash/activation-init.sh#L46-L52